### DISCUSSION TITLE
RPI Gpio

### DISCUSSION THREAD

**Discussion by stupidpersonthatisstupid** [OpenHantek/OpenHantek6022#163] [Ideas]
Hi, im wondering is it possible to have like external knobs and switches for the software. I know RPI has Gpio-s which could be used with like switches and incremental encoders. Also I could emulate key presses with the arduino and just imput them over usb to like use shortcuts and change stuff. So my question is has anyone done this and is it possible? 

Thanks in advance :)

**Comment by Ho-Ro** [comment]
Please provide a concept how to implement this in a generic way. I do not want to introduce regressions or additional support effort for the mainstream project caused by features that are only useful for a small user group.
Also I cannot support HW that is not on my desk.

**Comment by cryptozoy** [comment]
> Hi, im wondering is it possible to have like external knobs and switches for the software. I know RPI has Gpio-s which could be used with like switches and incremental encoders. Also I could emulate key presses with the arduino and just imput them over usb to like use shortcuts and change stuff. So my question is has anyone done this and is it possible?
> 
> Thanks in advance :)

It's not a problem. You can simply buy a conventional oscilloscope with controls, additionally equipped with a USB interface. But here only software is being developed specifically for Hantek 6022BE/BL and its analogues, and in the rare case, small hardware modifications to the oscilloscope itself.

**Comment by cryptozoy** [comment]
> Please provide a concept how to implement this in a generic way. I do not want to introduce regressions or additional support effort for the mainstream project caused by features that are only useful for a small user group.
> Also I cannot support HW that is not on my desk.

Do we need an API for external control of OpenHantek6022? Once created, anyone can bind anything to control the program. But is it really necessary?

**Comment by vizkoze** [comment]
Music programs like ableton can use midi controllers to bind Rotary encoders and buttons to specific functions of the program.
Maybe midi is the way to go or can provide a example.
________________________________

**Comment by stupidpersonthatisstupid** [comment]
> 
> 
> > Please provide a concept how to implement this in a generic way. I do not want to introduce regressions or additional support effort for the mainstream project caused by features that are only useful for a small user group.
> > Also I cannot support HW that is not on my desk.
> 
> Do we need an API for external control of OpenHantek6022? Once created, anyone can bind anything to control the program. But is it really necessary?

Well, nothing in life in necessary except breathing, eating and drinking but it would be really nice to have a battery powered scope with like knobs and stuff bc I like having an oscilloscope with a big screen and touchscreen is not the best for scope. My plan was to use the 6022BE with a rpi to run the the software and then have an external arduino which would reed all the knobs and buttons and then simulate key presses over usb. Basically I would have a nice battery powered scope with a nice big screen and all the buttons and stuff.

BDW. sorry for a late reply to this thread, I was super busy this week :)

**Comment by cryptozoy** [comment]
You can use this HScope application:
https://www.martinloren.com/hscope/
https://play.google.com/store/apps/details?id=com.martinloren.hscope
